plugins/woocommerce/includes/class-wc-customer.php

```diff
@@ -105,10 +113,12 @@ public function __construct( $data = 0, $is_session = false ) {
 			$this->set_id( $data );
 		}
 
-		$this->data_store = WC_Data_Store::load( 'customer' );
+		// We enable sessions if requested, the session handle exists, and the user is not logged in.
+		$this->is_session = $is_session && isset( WC()->session );
+		$this->data_store = $this->is_session ? WC_Data_Store::load( 'customer-session' ) : WC_Data_Store::load( 'customer' );
```

Comment:
Rather than use both data-stores (session being an incomplete store) we use one or the other depending on whether this is a session. The `customer-session` store will extend `customer`. The difference is, `customer-session` does not persist anything to the DB.